ui/src/app/views/rename-modal/component.ts

```diff
@@ -0,0 +1,24 @@
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+
+@Component({
+  selector: 'app-rename-modal',
+  styleUrls: ['./component.css',
+    '../../styles/buttons.css'],
+  templateUrl: './component.html',
+})
+export class RenameModalComponent {
+  public renaming = false;
+  public newName = '';
+
+  @Input() resourceType: string;
+  @Output() receiveRename = new EventEmitter<object>();
+  @Input() resource: any;
+
+  open(): void {
+    this.renaming = true;
+  }
+
+  close(): void {
```

Comment:
I think the pattern we've typically used is "submit modal" -> "load+spin" -> "close modal"

Reply:
OK, thanks - definitely leave as-is then.

As a follow-up we may want to address that behavior across the board, I'm not sure we want to close the dialog before the action is completed as things will shuffle around on the screen some indeterminate time after the modal has been closed. Alternatively we could have some indication that something is still loading somewhere else on the page.

Reply:
I used the same pattern as the ConfirmDeleteModal.... Maybe we should consider refactoring both? (Another ticket?)

Reply:
https://precisionmedicineinitiative.atlassian.net/browse/RW-1231